solutions/ha-nva/README.md

```diff
@@ -0,0 +1,244 @@
+
+### Simulated NVA VMs
+
+The simulated NVA VMs for this sample are CentOS 7 or RedHat 7 virtual machines running Apache httpd configured as a forward proxy on port 8080 to handle requests from the web-tier, and a reverse proxy to forward all requests on port 80 to the back end Load Balancer IP address (10.0.1.4). The script to set up the proxy is at: [scripts/simpleproxy.sh](scripts/simpleproxy.sh).
+
+### Simulated Web VMs
+
+The simulated web VMs for this sample are CentOS 7 or RedHat 7 virtual machines running Apache httpd with an index.html file generated to include the VM instance name from the Azure Instance Metadata Service. The script to set up the web server is at: [scripts/simpleweb.sh](scripts/simpleweb.sh).
+
+## Connecting to the web endpoint
+
+To test the ingress path, find the public IP address of the Application Gateway (ag-hanva) and enter it in the address bar of your web browser.  You should see a simple page indicating which back-end web server is providing the response. You can stop any of the NVA or web VMs, and the endpoint solution should automatically route requests through and to available VMs.
+
+## Connecting to deployed VMs with Azure Bastion
+
+An Azure Bastion service is deployed to provide access to the VMs in this architecture. The SSH Key required to connect to the VMs is generated at deployment time and stored in Azure Key Vault using the nested deployment script [nested/storekeyinkeyvault.json](nested/storekeyinkeyvault.json), which was adapted from [Create ssh-keys and store in Key Vault](https://azure.microsoft.com/resources/templates/201-deployment-script-ssh-key-gen/).
+
+In order to access the ssh key from the portal to use Azure Bastion, you will need to add an access policy on the Key Vault and grant yourself ***Secret*** **List** and **Get** permissions on `kv-hanva-{uniquestring}`.
+
+To test the egress path, connect to one of the web-tier VMs using Azure Bastion as described above. You can use a curl command like the following to confirm the egress path is working properly:
+
+```bash
+curl -v https://ifconfig.me
+```
+
+The output should show the connection to the internal NVA loadbalancer as well as the public IP address that is used to connect to the internet.  See the example below:
+
+```
+* About to connect() to proxy 10.0.0.68 port 8080 (#0) # Confirm proxy address is private IP of NVA Load Balancer
+*   Trying 10.0.0.68...
+* Connected to 10.0.0.68 (10.0.0.68) port 8080 (#0)
+> GET http://ifconfig.me/ HTTP/1.1
+> User-Agent: curl/7.29.0
+> Host: ifconfig.me
+> Accept: */*
+...
+< Via: 1.1 vm-nva-1.internal.cloudapp.net:8080 # Shows which NVA VM serviced the request
+<
+* Connection #0 to host 10.0.0.68 left intact
+203.0.113.34 # Should be the Public IP of the NVA VM
+```
+
+## Considerations
```

Comment:
Should we consider moving this content to the RA?

Reply:
The doc as it stands is more of a survey of all the different ways you can do HA for Network Virtual Appliances and the trade-offs and capabilities of each model.  A bunch of the content here in this README applies to 3 of the 5 architectures shown in the doc, so I decided to keep a bunch of the implementation details, consideration, and testing walkthrough in Github instead of trying to make it fit in the Doc.  I could see room for improvement in the doc as it stands, or possibly we could decompose it into different related articles with one article focused on the 3 layer-7 architectures, in which case more of the README would make sense in the doc.